src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -919,5 +919,80 @@ public static Task ReplayWarningsAndErrorsAsync(LockFile lockFile, ILogger logge
+        public static (ProjectStyle ProjectStyle, bool PackageReferenceCompatibleProjectStyle) GetProjectRestoreStyle(string restoreProjectStyle, bool hasPackageReferenceItems, string projectJsonPath, string projectDirectory, string projectName, ILogger log)
+        {
+            ProjectStyle projectStyle = ProjectStyle.Unknown;
+
+            // Allow a user to override by setting RestoreProjectStyle in the project.
+            if (!string.IsNullOrWhiteSpace(restoreProjectStyle))
+            {
+                if (!Enum.TryParse(restoreProjectStyle, out ProjectStyle userSuppliedProjectStyle))
+                {
+                    log.Log(LogMessage.CreateError(NuGetLogCode.NU1008, string.Format(CultureInfo.CurrentCulture, Strings.Error_InvalidProjectRestoreStyle, restoreProjectStyle)));
+                }
+
+                projectStyle = userSuppliedProjectStyle;
+            }
+            else if (hasPackageReferenceItems)
+            {
+                // If any PackageReferences exist treat it as PackageReference. This has priority over project.json.
+                projectStyle = ProjectStyle.PackageReference;
+            }
+            else if (!string.IsNullOrWhiteSpace(projectJsonPath))
+            {
+                // If this is not a PackageReference project check if project.json or projectName.project.json exists.
+                projectStyle = ProjectStyle.ProjectJson;
+            }
+            else if (MSBuildRestoreUtility.ProjectHasPackagesConfigFile(projectDirectory, projectName))
+            {
+                // If this is not a PackageReference or ProjectJson project check if packages.config or packages.ProjectName.config exists
+                projectStyle = ProjectStyle.PackagesConfig;
+            }
+            else
+            {
+                // This project is either a packages.config project or one that does not use NuGet at all.
+                projectStyle = ProjectStyle.Unknown;
+            }
+
+            bool packageReferenceCompatibleProjectStyle = projectStyle == ProjectStyle.PackageReference || projectStyle == ProjectStyle.DotnetToolReference;
+
+            return (projectStyle, packageReferenceCompatibleProjectStyle);
+        }
+
+        /// <summary>
+        /// Determines if the project has a packages.config file.
+        /// </summary>
+        /// <param name="projectDirectory">The full path of the project directory.</param>
+        /// <param name="projectName">The name of the project file.</param>
+        /// <returns><code>true</code> if a packages.config exists next to the project, otherwise <code>false</code>.</returns>
+        private static bool ProjectHasPackagesConfigFile(string projectDirectory, string projectName)
+        {
+            string packagesConfigPath = Path.Combine(projectDirectory, NuGetConstants.PackageReferenceFile);
+
+            if (File.Exists(packagesConfigPath))
```

Comment:
Do MSBuild `Exists(…)` checks like [this](https://github.com/NuGet/NuGet.Client/pull/3116/files#diff-3b0018f61173d0adf1ba163a23821381L439) and `File.Exists(…)` checks like this have identical behavior with respect to case-sensitivity (or case-insensitivity) on each platform?

I'm wondering if there is a case where one would have a different result than the other on a specific platform.

Reply:
MSBuild does not correct case either so this should behave the same